code/game/machinery/bots/ed209bot.dm

```diff
@@ -784,8 +784,8 @@ Auto Patrol: []"},
 	if(src.check_records)
 		for (var/datum/data/record/E in data_core.general)
 			var/perpname = perp.name
-			if(perp.wear_id)
-				var/obj/item/weapon/card/id/id = perp.wear_id.GetID()
+			if(perp.get_visible_id())
+				var/obj/item/weapon/card/id/id = perp.get_visible_id()
```

Comment:
You're running this proc twice then checking if it returned true twice. Just remove the `if(perp.get_visible_id())` and un-tab everything that was in it.